src/mlpack/core/cv/simple_cv.hpp

```diff
@@ -0,0 +1,131 @@
+/**
+ * @file simple_cv.hpp
+ * @author Kirill Mishchenko
+ *
+ * A simple cross-validation strategy.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_SIMPLE_CV_HPP
+#define MLPACK_CORE_CV_SIMPLE_CV_HPP
+
+#include <mlpack/core/cv/meta_info_extractor.hpp>
+#include <mlpack/core/cv/cv_base.hpp>
+
+namespace mlpack {
+namespace cv {
+
+/**
+ * The class SimpleCV splits data into training and validation sets, runs
+ * training on the training set and evaluates performance on the validation set.
+ *
+ * @tparam MLAlgorithm A machine learning algorithm.
+ * @tparam Metric A metric to assess the quality of a trained model.
+ * @tparam MatType The type of data.
+ * @tparam PredictionsType The type of predictions (should be passed when the
+ *     predictions type is a template parameter in Train methods of
+ *     MLAlgorithm).
+ * @tparam WeightsType The type of weights (should be passed when weighted
+ *     learning is supported, and the weights type is a template parameter in
+ *     Train methods of MLAlgorithm).
+ */
+template<typename MLAlgorithm,
+         typename Metric,
+         typename MatType = arma::mat,
+         typename PredictionsType =
+             typename MetaInfoExtractor<MLAlgorithm, MatType>::PredictionsType,
+         typename WeightsType =
+             typename MetaInfoExtractor<MLAlgorithm, MatType,
+                 PredictionsType>::WeightsType>
```

Comment:
Hmm, maybe there is something I have overlooked, but two questions here. Is it necessary to take the `MatType` parameter at the class level?  It seems like that should actually only be needed in `Evaluate()`, and could be deduced based on what the user passes in as arguments.  The second question is, is there any situation where `PredictionsType` and `WeightsType` cannot be directly derived from `MLAlgorithm` and `MatType`?  In the way this is set up, the user can set a custom `PredictionsType` and `WeightsType`, but I don't know if it is ever valid for them to be able to set anything other than what `MetaInfoExtractor` will provide.